src/frontend/screens/Library/components/SteamInstall/SteamInstallButton.tsx

```diff
@@ -0,0 +1,154 @@
+  const [showInstallDialog, setShowInstallDialog] = useState(false)
+  const [isCompatibilityLayerAvailable, setIsCompatibilityLayerAvailable] =
+    useState(false)
+  const [showAlert, setShowAlert] = useState<'success' | 'danger' | 'none'>(
+    'none'
+  )
+
+  const isMac = platform === 'darwin'
+  const isSteamInstalled =
+    isMac &&
+    libraryState.sideloadedLibrary.find((game) => game.app_name === 'steam')
+
+  const { t } = useTranslation()
+  useEffect(() => {
+    const checkWine = async () => {
+      const winelist: WineInstallation[] = await window.api.getAlternativeWine()
+      if (winelist.length > 0) {
+        setIsCompatibilityLayerAvailable(true)
+      }
+    }
+    checkWine()
+  }, [])
+
+  async function handleSteamInstallation() {
+    if (isSteamInstalled) {
+      return launch({
+        appName: 'steam',
+        t,
+        runner: 'sideload',
+        showDialogModal,
+        hasUpdate: false,
+        isNotNative: true
+      })
+    }
+
+    return setShowInstallDialog(true)
+  }
+
+  function renderButtonText() {
+    if (isInstalling) {
+      return t('Installing Steam', 'Installing Steam...')
+    }
+    if (isSteamInstalled) {
+      return t('Launch Steam', 'Launch Steam')
+    }
+    return t('Install Steam', 'Install Steam')
+  }
+
+  const toolTipText = isSteamInstalled
+    ? t(
+        'steam-install.launch-tooltip',
+        'Launch Steam using the HyperPlay Compatibility Layer'
+      )
+    : t(
+        'steam-install.tooltip',
+        `This will install the Windows version of Steam and it will run using a HyperPlay Compatibility Layer. \n
+    Some games especially older titles, newer 3D games, or those with anti-cheat software may not work properly. \n
+    The Recommended specs are a Mac with Apple Silicon CPU (M1 or above) and at least 16GB of RAM.`
+      )
+
+  const comatibilityLayerNotAvailableTooltip = t(
+    'steam-install.compatibility-layer-not-available',
+    `Compatibility layer not available. Please install one from "Settings > Wine Manager" first.`
+  )
+
+  const showSuccessAlert = showAlert === 'success'
+  const showErrorAlert = showAlert === 'danger'
+
+  return (
+    <>
+      <Tooltip
+        {...tooltipProps}
+        label={
+          isCompatibilityLayerAvailable
+            ? toolTipText
+            : comatibilityLayerNotAvailableTooltip
+        }
+        className={'Tooltip caption-sm'}
+      >
+        <Button
+          type="tertiary"
+          onClick={handleSteamInstallation}
+          leftIcon={<FontAwesomeIcon icon={faSteam} height={14} width={14} />}
+          disabled={!isCompatibilityLayerAvailable || isInstalling}
+        >
+          {renderButtonText()}
+        </Button>
+      </Tooltip>
+      {showInstallDialog ? (
+        <SteamInstallDialog
+          isInstalling={isInstalling}
+          setIsInstalling={setIsInstalling}
+          onClose={() => setShowInstallDialog(false)}
+          setShowAlert={setShowAlert}
+        />
```

Comment:
I think the implementation should be:

```tsx
<SteamInstallDialog
    isInstalling={isInstalling}
    onInstall={installSteamMutation.mutate}
    onClose={() => setShowInstallDialog(false)}
/>
```

you can do the alert if you move the mutation to this component.

Reply:
no need to pass the state setters